rs/nns/governance/src/neuron/types/tests.rs

```diff
@@ -451,3 +452,88 @@ fn test_neuron_configure_dissolve_delay() {
     let now = now + 1;
     assert_eq!(neuron.state(now), NeuronState::Dissolved);
 }
+
+#[test]
+fn test_visibility_when_converting_neuron_to_neuron_info_and_neuron_proto() {
+    // (These are not actually used by code under test.)
+    let principal_id = PrincipalId::new_user_test_id(42);
+    let timestamp_seconds = 1729791574;
+
+    let builder = NeuronBuilder::new(
+        NeuronId { id: 42 },
+        Subaccount::try_from(vec![42u8; 32].as_slice()).unwrap(),
+        principal_id,
+        DissolveStateAndAge::NotDissolving {
+            dissolve_delay_seconds: 1_000_000,
+            aging_since_timestamp_seconds: timestamp_seconds,
+        },
+        timestamp_seconds, // created
+    );
+
+    // Case 1: visibility is explicitly set.
+    for set_enforcement in [
+        temporarily_enable_private_neuron_enforcement,
+        temporarily_disable_private_neuron_enforcement,
+    ] {
+        let _restore_on_drop = set_enforcement();
+
+        for visibility in [Visibility::Public, Visibility::Private] {
+            let neuron = builder.clone().with_visibility(Some(visibility)).build();
+
+            assert_eq!(neuron.visibility(), Some(visibility),);
+
+            let neuron_info = neuron.get_neuron_info(timestamp_seconds, principal_id);
+            assert_eq!(neuron_info.visibility, Some(visibility as i32),);
```

Comment:
Nit. I would slightly prefer to use `Visibility::from(i32)` throughout this PR instead of `visibility as i32`. It's up to you to decide which way to go, as this is cosmetic.

Reply:
Let's do this opportunistically / in a follow-up PR; IMHO we shouldn't block the release process on this. Just wanted to point this out as a best practice suggestion.